resource-managers/kubernetes/core/src/main/scala/org/apache/spark/scheduler/cluster/k8s/ExecutorPodsAllocator.scala

```diff
@@ -214,10 +216,9 @@ private[spark] class ExecutorPodsAllocator(
     }
 
     if (newlyCreatedExecutors.isEmpty
-        && currentPendingExecutors.isEmpty
-        && currentRunningCount < currentTotalExpectedExecutors) {
+        && knownPodCount < currentTotalExpectedExecutors) {
```

Comment:
This is irrelevant to `[SPARK-33231][CORE] Make pod allocation executor timeouts configurable`. Could you make another PR about this logic change?

Reply:
@holdenk . Please split this part.

Reply:
So it's not unrelated, but I can give it a second JIRA? Having more time available means that depending on currentRunningCount can be stale information.